src/participant/participantTypes.ts

```diff
@@ -0,0 +1,13 @@
+import type * as vscode from 'vscode';
+
+export type SendMessageToParticipantOptions = {
+  message: string;
+  isNewChat?: boolean;
+  isPartialQuery?: boolean;
+};
+
+export type SendMessageToParticipantFromInputOptions = {
+  messagePrefix?: string;
+  source?: 'query with copilot codelens';
```

Comment:
```suggestion
  source?: string;
```
Can it be maybe more generic and just a string?

Reply:
I agree that this should be flexible but I was thinking of this as documentation of the sources we have to prevent inconsistencies, like an enum, so in the future if we have more we can add more. I could make a literal enum similar to `DocumentSource`

Reply:
Yeah, let's make it maybe a part of DocumentSource then.